enterprise/views.py

```diff
@@ -1066,15 +1066,21 @@ def get(self, request):
             )
             query_dict['next'] = learner_portal_base_url + '/' + enterprise_slug
 
-        if enterprise_customer.identity_provider:
+        tpa_hint = query_params.get('tpa_hint')
+        if tpa_hint:
+            # If there is tpa_hint already in the query_string then no need to update it in query_dict
+            pass
+        elif enterprise_customer.identity_providers and enterprise_customer.identity_providers.count() == 1:
```

Comment:
do we need non-null check on `enterprise_customer.identity_providers` ? shouldn't  `enterprise_customer.identity_providers.count() == 1` be enough?

Reply:
Wouldn't we need that check if `enterprise_customer.identity_providers` is `None`?

Reply:
Or will this property return an empty queryset in that case?

Reply:
It will return an empty query set if there are no associated identity providers.